aziot/src/main.rs

```diff
@@ -7,48 +7,43 @@
     clippy::let_unit_value,
     clippy::similar_names,
     clippy::too_many_lines,
-    clippy::type_complexity
+    clippy::type_complexity,
+    clippy::clippy::module_name_repetitions
 )]
 
+use anyhow::Result;
+use structopt::StructOpt;
+
+mod internal;
+
+mod check;
+mod check_list;
 mod init;
 
-fn main() -> Result<(), Error> {
-    let options = structopt::StructOpt::from_args();
+async fn try_main() -> Result<()> {
+    let options = StructOpt::from_args();
     match options {
         Options::Init => init::run()?,
+        Options::Check(cfg) => check::check(cfg).await?,
+        Options::CheckList(cfg) => check_list::check_list(cfg)?,
     }
 
     Ok(())
 }
 
-#[derive(structopt::StructOpt)]
-enum Options {
-    Init,
-}
-
-struct Error(Box<dyn std::error::Error>, backtrace::Backtrace);
-
-impl std::fmt::Debug for Error {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        writeln!(f, "{}", self.0)?;
-
-        let mut source = self.0.source();
-        while let Some(err) = source {
-            writeln!(f, "caused by: {}", err)?;
-            source = err.source();
-        }
-
-        writeln!(f, "{:?}", self.1)?;
-
-        Ok(())
+#[tokio::main]
+async fn main() {
+    if let Err(err) = try_main().await {
+        eprintln!("{:?}", err);
```

Comment:
`"{}"`, no?

Reply:
Does it print `caused by` chains the same way as other crates in the repo (and the checks in verbose mode) do it manually?

Reply:
Ah, yes, I was too quick on the trigger switching from `{:?}` to `{}`. See the corresponding `anyhow` docs: https://docs.rs/anyhow/1.0.37/anyhow/struct.Error.html#display-representations

I'll revert back to `{:?}`

Reply:
But even that is not identical to the way the rest of the code prints cause-chains and backtraces, which was my point. anyhow's cause chain appears to use `Caused by:\n    ` as opposed to the rest of the code's `caused by: `, and its backtrace appears to use `Stack backtrace:` as opposed to `backtrace`'s `backtrace:`

It would be good to be consistent, ie either format anyhow's error manually, or change the rest of the code to match anyhow's style.

Reply:
guh - I'm not sure if that's something I'd want to spend more time on.

I agree that consistency would be nice, but given that the `aziot` binary is totally distinct from `aziotd`, both in function and in its core error handling abstraction, I think it's fine leaving it as-is for now. If the inconsistency bothers you that much, feel free to tweak `aziotd`'s formatting in a separate PR.

Reply:
>feel free to tweak `aziotd`'s formatting in a separate PR.

Yes, that's fine.